### PR TITLE
Remove state machine 

### PR DESCRIPTION
**Pull request by wero1414**
Removed the state machine to get better energy consuption

### CODE REVIEW

File: src/arduino-rfm/LoRaMAC.cpp [hunk 1]
```diff
@@ -616,21 +730,25 @@ bool LORA_join_Accept(sBuffer *Data_Rx,sLoRa_Session *Session_Data, sLoRa_OTAA *
...
 				joinStatus = true;
 			}
 		}
 	}
 	return joinStatus;
 }
 
+void LORA_Set_Interrupt(void){
+    pinMode(RFM_pins.DIO0, INPUT);
+    RFM_Write(0x12, 0x00); //REG_DIO_MAPPING_1 RX done
```
Comment (sabas1080) [inline src/arduino-rfm/LoRaMAC.cpp]: Add new constants names "RFM_REG_IRQ_FLAGS"

File: src/arduino-rfm/lorawan-arduino-rfm.cpp [hunk 2]
```diff
@@ -242,58 +229,80 @@ void LoRaWANClass::setDevAddr(const char *devAddr_in)
...
 void LoRaWANClass::setDeviceClass(devclass_t dev_class)
 {
     LoRa_Settings.Mote_Class = (dev_class == CLASS_A)? CLASS_A : CLASS_C;
 
-    if (LoRa_Settings.Mote_Class == CLASS_A) {
-        RFM_Switch_Mode(RFM_MODE_STANDBY);
-    } else {
-        RFM_Continuous_Receive(&LoRa_Settings);
-    }
+    if (LoRa_Settings.Mote_Class == 0x00) {
```
Comment (sabas1080) [inline src/arduino-rfm/lorawan-arduino-rfm.cpp]: I think we should choose define CLASS_A AND CLASS_C or the value hexadecimal and comment, here is hexadecimal

File: src/arduino-rfm/lorawan-arduino-rfm.cpp [hunk 3]
```diff
@@ -242,58 +229,80 @@ void LoRaWANClass::setDevAddr(const char *devAddr_in)
...
     //Reset frame counter
     Frame_Counter_Tx = 0x0000;
-
-    //Reset RFM command status
-    RFM_Command_Status = NO_RFM_COMMAND;
 }
 
 void LoRaWANClass::setDeviceClass(devclass_t dev_class)
 {
     LoRa_Settings.Mote_Class = (dev_class == CLASS_A)? CLASS_A : CLASS_C;
```
Comment (sabas1080) [inline src/arduino-rfm/lorawan-arduino-rfm.cpp]: CLASS_A or CLASS C? or hexadecimal?? 🤔

File: src/arduino-rfm/LoRaMAC.cpp (hunk 1 above)
Comment (wero1414) [reply src/arduino-rfm/LoRaMAC.cpp]: Almost all the registers are used without variable names

File: src/arduino-rfm/lorawan-arduino-rfm.cpp (hunk 3 above)
Comment (wero1414) [reply src/arduino-rfm/lorawan-arduino-rfm.cpp]: Is the same, but i should be using only the CLASS_A or CLASS_C typos

File: src/arduino-rfm/LoRaMAC.cpp (hunk 1 above)
Comment (mr3188) [reply src/arduino-rfm/LoRaMAC.cpp]: Did you created names for the registers hex values? I have already created most of them in my repo and don't know if it's better to create a pull request now or wait until you merge this one.

Comment (wero1414) [reply src/arduino-rfm/LoRaMAC.cpp]: The better way should be by doing a pull request so you add all the registers numbers

**Pull request closed without merging**